paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/PaparazziViewOwners.kt

```diff
@@ -0,0 +1,48 @@
+package app.cash.paparazzi.internal
+
+import android.view.View
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import androidx.savedstate.SavedStateRegistry
+import androidx.savedstate.SavedStateRegistryController
+import androidx.savedstate.SavedStateRegistryOwner
+import androidx.savedstate.setViewTreeSavedStateRegistryOwner
+
+internal object PaparazziViewOwners {
+  private val hasLifecycleOwnerRuntime = isPresentInClassPath("androidx.lifecycle.LifecycleOwner")
+  private val hasSavedStateRegistryOwnerRuntime = isPresentInClassPath("androidx.savedstate.SavedStateRegistryOwner")
+
+  fun register(view: View) {
+    if (hasLifecycleOwnerRuntime) {
+      val lifecycleOwner = PaparazziLifecycleOwner()
+      ViewTreeLifecycleOwner.set(view, lifecycleOwner)
+
+      if (hasSavedStateRegistryOwnerRuntime) {
+        view.setViewTreeSavedStateRegistryOwner(PaparazziSavedStateRegistryOwner(lifecycleOwner))
+      }
+    }
+  }
+
+  private fun isPresentInClassPath(className: String): Boolean {
+    return try {
+      Class.forName(className)
+      true
+    } catch (e: ClassNotFoundException) {
+      false
+    }
+  }
+}
+
+private class PaparazziLifecycleOwner : LifecycleOwner {
+  private val lifecycleRegistry = LifecycleRegistry(this).apply { currentState = Lifecycle.State.CREATED }
+  override fun getLifecycle(): Lifecycle = lifecycleRegistry
+}
+
+private class PaparazziSavedStateRegistryOwner(
+  private val lifecycleOwner: LifecycleOwner
```

Comment:
does this need the lifecycle owner?

Reply:
Yes. `SavedStateRegistryOwner` extends `LifecycleOwner` so it is dependent. `OnBackPressedDispatcher` will have the same dependency.